src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java

```diff
@@ -80,18 +80,22 @@ private static LfBranchImpl createTransformer(TwoWindingsTransformer twt, LfBus
                 List<PiModel> models = new ArrayList<>();
 
                 for (int position = ptc.getLowTapPosition(); position <= ptc.getHighTapPosition(); position++) {
-                    PhaseTapChangerStep step = ptc.getStep(position);
-                    double r = twt.getR() * (1 + step.getR() / 100) / zb;
-                    double x = twt.getX() * (1 + step.getX() / 100) / zb;
-                    double g1 = twt.getG() * (1 + step.getG() / 100) * zb;
-                    double b1 = twt.getB() * (1 + step.getB() / 100) * zb;
-                    double r1 = twt.getRatedU2() / twt.getRatedU1() * ptc.getCurrentStep().getRho() / nominalV2 * nominalV1;
-                    double a1 = Math.toRadians(step.getAlpha());
+                    ptc.setTapPosition(position);
```

Comment:
By setting the current tap position, you change the input network and you lost this information of the tap position.
We probably should use methods like Transformers.getG1(twt, position, specificCompatibility) so with an additional position parameter to not change the network.